[ruby-2.3 feature #10769] add Enumerable#chunk_while, spec for it

cthulhua opened the pull request:
Implemented as #slice_when, but slice when block does not match rather than when it does.
Also added merged over ruby spec for it.

kares commented:
@eregon is it fine to have the rubyspec updates coming in like they are here?

cthulhua commented:
I saw that a similar thing was done for slice_when, so I added them.

eregon commented:
@kares @cthulhua Sorry, I missed your mention.
It would be better and simpler (for merging) to just let the regular rubyspec import take care of it.
But if it's the exact same change, then most likely there won't be a conflict (to be verified though).

cthulhua commented:
Ah OK - would you prefer that I remove the change? I wrote the rubyspec for this so I know there is no conflict - seems more hygienic to let the standard ruby spec import handle it though.

eregon commented:
@cthulhua If it's not much effort, this would be cleaner :smiley: 

No, we should just import latest rubyspec into this repo before the 2.3 merge
(@kares you can ask me, otherwise I do it every month).

cthulhua commented:
@eregon I removed the rubyspec change. I noticed that in ruby/ruby there was a test added for chunk_while (https://github.com/ruby/ruby/blob/trunk/test/ruby/test_enum.rb#L683) so i added it to the same place under test/mri, since that seemed to be what other folks did when they added ruby-2.3 features.

eregon commented:
Thanks :+1:
